.github/workflows/test.yml

```diff
@@ -1,25 +1,40 @@
 name: Tests
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - "master"
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    branches:
+      - "master"
 
 jobs:
   test:
     name: Test
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
-      - name: Prepare
+      - name: Install Dependencies
         run: |
-          sudo apt-get update
-          sudo apt install libfuse2
-          sudo add-apt-repository universe
-          wget https://github.com/neovim/neovim/releases/download/nightly/nvim.appimage
-          chmod u+x nvim.appimage
-          sudo mv nvim.appimage /usr/bin/nvim
-          git clone https://github.com/nvim-lua/plenary.nvim.git ~/.local/share/nvim/site/pack/plenary.nvim/start/plenary.nvim
+          mkdir -p ~/.local/share/nvim/site/pack/plenary.nvim/start
```

Comment:
I have a good `init.lua` for testing that install the dependencies in the init instead. It makes it easier to provide a minimal init that boostraps itself.

https://github.com/ten3roberts/recipe.nvim/blob/main/tests/init.lua

Perhaps we could have that instead later

Reply:
Sure, I'm open to whatever in that realm - it's just that the current setup fails for me, and the one I proposed works. Your `init.lua` looks great (even if the indentation is... excessive :P )- I'd be happy to integrate that instead, after.

Reply:
Oh, that's just github rendering tabs with 8 spaces for some reason.

Have yet to figure out if you should use spaces or tabs for lua, as *stylua* seems to insert tabs.

I see, do you refer to the plenary library not being found?

Reply:
Ahhh, ok. As much as I like the idea of tabs, coming from ruby I'm happy to stick with 2 spaces. I have `.editorconfig` set to do spaces, and stylua seems to respect that.

Yeah, because by default `make test` uses your config and presumes you already have plenary installed. I don't want it using _my_ config, since... thats a bad test. My personal modifications shouldn't impact the tests, you know? I definitely like your idea to overload the `XDG_*` variables. I had the same goal by passing the `NVIM_APPNAME` and git config variables. I think either approach would work, as long as personal config doesn't leak into tests :)